downstream/_auxlib/_unpack_hex.py

```diff
@@ -37,16 +45,25 @@ def unpack_hex(hex_str: str, num_items: int) -> np.ndarray:
         ascii_codes = np.fromstring(hex_str.lower(), dtype="S1", sep="").view(
             np.uint8,
         )
+
         digits = ascii_codes - ord("0")
         alphas = ascii_codes - (ord("a") - 10)
-        return np.where(digits < 10, digits, alphas)
+        result = np.where(digits < 10, digits, alphas)
+
+        if byteorder == "little":
+            result[::2] = result[-2::-2]
+            result[1::2] = result[::-2]
+        return result
 
     # unpack hex string into numpy bytes array
     bytes_array = np.frombuffer(
         bytes.fromhex(hex_str),
         count=len(hex_str) >> 1,
         dtype=np.uint8,
     )
+    if byteorder == "little":
```

Comment:
Are we sure we should be reversing the order of the whole array? The endianness should only affect the byte order *within* individual values (I think?)